### PR TITLE
[WIP] [CI] graph break on  c-tor functions

### PR DESCRIPTION
**Pull request by Krovatkin**
Fixes #ISSUE_NUMBER

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by Krovatkin** [comment]
If we let inductor create tensors, it won't set `requires_grad` on `empty_strided` calls in https://github.com/pytorch/pytorch/blob/master/torch/_inductor/codegen/wrapper.py#L51. This attribute is also missing on buffers created by tensor c-tors. So we end up creating tensors with this attribute still set to False so `backward()` doesn't compute gradients for those tensors (i.e. `.grad`) so when we try to access `.grad` we fail in the test.

Horace argues that torchdynamo is supposed to graph break on tensor constructors and this isn't an inductor issue.
